src/nvim/drawscreen.c

```diff
@@ -1516,10 +1516,12 @@ static void win_update(win_T *wp)
 
   decor_providers_invoke_win(wp);
 
-  if (win_redraw_signcols(wp)) {
-    wp->w_lines_valid = 0;
-    wp->w_redr_type = UPD_NOT_VALID;
-    changed_line_abv_curs_win(wp);
+  FOR_ALL_WINDOWS_IN_TAB(win, curtab) {
```

Comment:
Should this be now in `update_screen()` instead?

Reply:
Yeah agree that seems to more natural place for it now but that wouldn't solve the linked issue. It needs to happen after `on_win` callbacks.